src/utils/process_utils/src/process_utils.cpp

```diff
@@ -21,6 +19,9 @@
 #include <aduc/config_utils.h>
 #include <aduc/logging.h>
 #include <aduc/string_utils.hpp>
+
+#include <aducpal/stdio.h> // popen,pclose
```

Comment:
super nit: needs ADUCPAL_ prefixes or omit

Reply:
my comments indicate the underlying api.